crates/celo-revm/src/api/celo_block_env.rs

```diff
@@ -3,11 +3,17 @@ use op_revm::L1BlockInfo;
 use revm::primitives::Log;
 use std::vec::Vec;
 
+#[derive(Debug, Clone, Default)]
+pub struct Cip64Info {
+    pub actual_intrinsic_gas_used: u64,
+    /// Logs from system calls (debit/credit) that need to be merged into the final receipt
+    pub logs_pre: Vec<Log>,
+    pub logs_post: Vec<Log>,
+}
+
 #[derive(Debug, Clone, Default)]
 pub struct CeloBlockEnv {
     pub l1_block_info: L1BlockInfo,
     pub fee_currency_context: FeeCurrencyContext,
-    /// Logs from system calls (debit/credit) that need to be merged into the final receipt
-    pub cip64_actual_tx_system_call_logs_pre: Vec<Log>,
-    pub cip64_actual_tx_system_call_logs_post: Vec<Log>,
+    pub actual_cip64_tx_info: Cip64Info,
```

Comment:
Maybe we can find a tx-related place to put this information instead of the block env. I'll see if I find something to suggest.

Reply:
Maybe as a new field in the CeloTransaction?

Reply:
I was able to do it. As the tx is not mutable in the context, I had to clone and set it a few times, but I think it keeps everything cleaner